src/cdk/table/table.spec.ts

```diff
@@ -356,6 +358,62 @@ describe('CdkTable', () => {
     ]);
   });
 
+  it('should be able to use the array data source to render the table', () => {
+    const arrayCdkTableFixture = TestBed.createComponent(ArrayDataSourceCdkTableApp);
+    tableElement = arrayCdkTableFixture.nativeElement.querySelector('cdk-table');
+    arrayCdkTableFixture.detectChanges();
+
+    expectTableToMatchContent(tableElement, [
+      ['Column A', 'Column B', 'Column C'],
+      ['a_1', 'b_1', 'c_1']
+    ]);
+
+    const arrayDataSource = arrayCdkTableFixture.componentInstance.dataSource;
+    arrayDataSource.data = [
+      {a: 'a_1', b: 'b_1', c: 'c_1'},
+      {a: 'a_2', b: 'b_2', c: 'c_2'},
+      {a: 'a_3', b: 'b_3', c: 'c_3'},
+    ];
+    arrayCdkTableFixture.detectChanges();  // Allow cells to populate their templates
+
+    expectTableToMatchContent(tableElement, [
+      ['Column A', 'Column B', 'Column C'],
+      ['a_1', 'b_1', 'c_1'],
+      ['a_2', 'b_2', 'c_2'],
+      ['a_3', 'b_3', 'c_3'],
+    ]);
+
+    // Making a change to the data should update the table.
+    arrayDataSource.data[0].a = 'changed_a_1';
+    arrayCdkTableFixture.detectChanges();  // Allow cells to populate their templates
+    expectTableToMatchContent(tableElement, [
+      ['Column A', 'Column B', 'Column C'],
+      ['changed_a_1', 'b_1', 'c_1'],
+      ['a_2', 'b_2', 'c_2'],
+      ['a_3', 'b_3', 'c_3'],
+    ]);
+
+    // Adding a new object will not update the table until update called.
+    arrayDataSource.data.push({a: 'a_4', b: 'b_4', c: 'c_4'});
+    arrayCdkTableFixture.detectChanges();  // Prove that detect changes won't render row
+    expectTableToMatchContent(tableElement, [
+      ['Column A', 'Column B', 'Column C'],
+      ['changed_a_1', 'b_1', 'c_1'],
+      ['a_2', 'b_2', 'c_2'],
+      ['a_3', 'b_3', 'c_3'],
+    ]);
+
+    arrayDataSource.update();  // Calling update will trigger update
```

Comment:
I removed this line and the test still passed. I'm afraid there is a bug in `expectTableToMatchContent`